templates/bin_setup

```diff
@@ -13,7 +13,10 @@ bundle check || bundle install
 # Install JavaScript dependencies
 bin/yarn
 
-# Set up database and add any development seed data
+# Set up database and run migrations
+bin/rails db:reset || bin/rails db:migrate
```

Comment:
This is inspired by [David Copeland's Rails template](https://github.com/davetron5000/rails-app-template-sustainable/blob/main/bin/setup.tt#L20-L23).

Basically, we want to be able to confidently run migrations without needing to know if the database exists first.

Reply:
Is there a situation where `bin/rails db:reset` would fail? If I do `bin/rails db:drop` and then `bin/rails db:reset` there's no failure -- if the database does not exist, it still displays the message "Dropped database ....".

I wonder if that was an issue in older Rails versions. If `db:reset` consistently works everytime, I'd drop the `|| bin/rails db:migrate`. What do you think?

Reply:
Never mind, I think I understand the purpose. I will delete this comment and write another one :)

Reply:
@stevepolitodesign Maybe `db:reset` would be too destructive? I would not expect `bin/setup` to destroy and recreate my database, but rather to be idempotent. What do you think of:

```sh
bin/rails db:setup || bin/rails db:migrate
```

`db:setup` doesn't run `db:drop`, so it may be a better option.

If `db:setup` fails, `db:migrate` will be run as a fallback. `db:reset`, on the other hand, will always drop the database so it would probably never fallback to `db:migrate`.

> There might be some room for optimizations since running bin/rails db:reset drop and creates the database, and then bin/rails dev:prime does the same thing by invoking db:setup

Maybe you could update `dev.rake` with something like this to DRY it up:

```ruby
if Rails.env.development? || Rails.env.test?
  require "factory_bot"

  namespace :dev do
    task sync_db: :environment do
      `bin/rails db:setup || bin/rails db:migrate`
    end

    desc "Sample data for local development environment"
    task prime: "dev:sync_db" do
      include FactoryBot::Syntax::Methods

      # create(:user, email: "user@example.com", password: "password")
    end
  end
end
```

The `sync_db` task would be hidden because there's no `desc`, and it wouldn't appear in `rake -T`.